tests/gen_cert_test.py

```diff
@@ -181,3 +179,69 @@ def test_cert_date(self, force_date, locale, timezone, expected):
         """
         cert_date = get_cert_date(force_date, locale, timezone)
         assert_equal(expected, cert_date)
+
+
+class CertificateSubtemplateTests(unittest.TestCase):
+    """
+    Tests the subtemplates feature for certs
+
+    Subtemplates are currently written to key on designations only
+
+    These tests have a specific order
```

Comment:
Yes, @ggg, that is the relevant question here.   Thanks for looking at the code. 

It was helpful when writing and debugging to have them in separate functions.  It's also easier to identify what is failing when running tests; Each tests function tests one thing.  However, I am also uncomfortable with them being in ordered, hence my adding the the comment in the first place.  

Given the current state of tests for this repo I'm okay with CertificateSubtemplateTests as is.  Really, the whole test file could be overhauled.  I'm also not convinced this feature has fully evolved.  We already have plans for removing designations for example.  Also, This feature could get more complex and we could have a need for multiple test course_ids.  

Related side note: 
Looping over multiple certs and having them in separate functions are not related.  We could remove the loop and still have three function or keep the loop and merge them into one.